### PR TITLE
add regex path

### LINKED ISSUE #164: Accommodate Regex Routes

### PR DESCRIPTION
**Pull request by NorthFury**
closes #164

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala [hunk 1]
```diff
@@ -54,12 +54,19 @@ class DefaultRouteToEskipTransformer @Inject() (hostsService: HostsService, comm
...
   }
 
-  private[this] def createPathPredicate(pathUri: String, hasStar: Boolean) = {
-    val starSuffix = if (hasStar) "/**" else ""
+  private val STAR_PATH_SUFFIX = "/**"
+  private[this] def createPathPredicate(pathUri: String, hasStar: Boolean, isRegex: Boolean) = {
+    val pathArg = (hasStar, isRegex) match {
+      case (false, false) => wrapInQuotes(pathUri)
+      case (true, false)  => wrapInQuotes(pathUri + STAR_PATH_SUFFIX)
+      case (_, true)      => pathUri
```
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala]: don't we wrap the regex in `/^...$/`

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala [hunk 2]
```diff
@@ -224,6 +249,18 @@ class PostPathsSpec extends FunSpec with BeforeAndAfter with Matchers {
           entityString(response).parseJson.convertTo[Error].errorType should be("ITE")
         }
       }
+
+      describe("when a token without admin privileges is provided when creating a regex path") {
```
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala]: does this case cover the admin team?

Comment (NorthFury) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala]: good point, I'll push a test for the admin team

File: src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala (hunk 1 above)
Comment (NorthFury) [inline src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala]: I made the assumption that the slashes are in the string. I guess it is better to only have the regex expression. I'll change it.

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala (hunk 2 above)
Comment (NorthFury) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala]: This test is for a token without admin privileges. A new test isn't needed for the fail path. One more test is needed for the happy path.

Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/PostPathsSpec.scala]: 👍

File: src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala (hunk 1 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/services/RouteToEskipTransformer.scala]: please check with @aryszka what's the best approach

**Comment by danpersa** [comment]
👍

**Pull request merged**